src/lib/lexer.rs

```diff
@@ -81,16 +81,73 @@ impl<TokId: Copy + Eq> LexerDef<TokId> {
     }
 
     /// Set the id attribute on rules to the corresponding value in `map`. This is typically used
-    /// to synchronise a parser's notion of token IDs with the lexers.
-    pub fn set_rule_ids(&mut self, map: &HashMap<&str, TokId>) {
-        for r in self.rules.iter_mut() {
+    /// to synchronise a parser's notion of token IDs with the lexers. While doing this, it keeps
+    /// track of which tokens:
+    ///   1) are defined in the lexer but not referenced by the parser
+    ///   2) and referenced by the parser but not defined in the lexer
+    /// and returns them as a tuple `(Option<HashSet<&str>>, Option<HashSet<&str>>)` in the order
+    /// (defined_in_lexer_missing_from_parser, referenced_in_parser_missing_from_lexer). Since in
```

Comment:
Ah, I see. I was expecting to see those names in the block below! Maybe make them italic?

```markdown
(*defined-in-lexer-missing-from-parser*, *referenced-in-parser-missing-from-lexer*)
```